src/components/TextField/TextField.tsx

```diff
@@ -0,0 +1,247 @@
+    e.preventDefault();
+    e.stopPropagation();
+
+    const inputElem = getInitializedRef(inputRef);
+    const lengthCountElem = getInitializedRef(lengthCountRef);
+
+    inputElem.placeholder = placeholder || '';
+    inputElem.value = '';
+    lengthCountElem.innerText = getStringOfValueLengthPerMax('', max);
+    onChange(e);
+    setTextFieldStatus('default');
+  };
+
+  const handleBlur: FocusEventHandler<HTMLInputElement> = e => {
+    const inputElem = getInitializedRef(inputRef);
+
+    if (inputElem.value.length <= 0) inputElem.placeholder = placeholder || '';
+
+    onBlur(e);
+    updateStatusFromInputValue(inputElem.value);
+  };
+
+  const handleFocus: FocusEventHandler<HTMLInputElement> = () => {
+    const inputElem = getInitializedRef(inputRef);
+
+    if (inputElem.value === '') inputElem.placeholder = '';
+
+    updateStatusFromInputValue(inputElem.value);
+  };
+
+  const handleInputChange: ChangeEventHandler<HTMLInputElement> = async e => {
+    const inputElem = getInitializedRef(inputRef);
+    const lengthCountElem = getInitializedRef(lengthCountRef);
+
+    lengthCountElem.innerText = getStringOfValueLengthPerMax(
+      inputElem.value,
+      max
+    );
+
+    onChange(e);
+    const valdationResult = await validate(inputElem.value);
+
+    if (valdationResult.result === true) return setTextFieldStatus('active');
+    if (valdationResult.type === 'max')
+      return setTextFieldStatus('error', '글자수를 초과했습니다.');
+    if (valdationResult.type === 'email')
+      return setTextFieldStatus('error', '이메일 형식이 아닙니다.');
+  };
+
+  return (
+    <div
+      className={clsx([style.inputTypeRecipe({ status }), style.wrapper])}
+      arial-lable="text"
+    >
+      <label className={clsx([style.label, variants.caption1])}>{label}</label>
+      <div className={style.inputContainer}>
+        <input
+          name={name}
+          type={type}
+          className={style.input({ size })}
+          placeholder={placeholder}
+          onChange={handleInputChange}
+          onBlur={handleBlur}
+          onFocus={handleFocus}
+          ref={node => {
+            inputRef.current = node; // TextField 내부의 inputRef 사용
+            if (typeof ref === 'function') ref(node); // RefCallback이 오는 경우 적용
+          }}
+        />
+        <RemoveButton
+          onClick={handleRemoveClick}
+          inputRef={inputRef}
+          status={status}
+        />
+        <Count max={max} ref={lengthCountRef} />
+        <div className={style.underbar} />
+      </div>
+      <Message status={status} message={message} />
+    </div>
+  );
+});
+export default TextField;
+
+/**
+ *
+ */
+const RemoveButton = ({
+  onClick,
+  inputRef
+}: {
+  onClick: MouseEventHandler<HTMLButtonElement>;
+  inputRef: { current: HTMLInputElement | null };
+  status: TextFieldStatus;
+}) => {
+  const checkIsOnOff = () =>
+    inputRef.current && inputRef.current.value.length && status !== 'default'
+      ? 'on'
+      : 'off';
```

Comment:
위 코멘트와 이어집니다
style.icon의 variant를 boolean값으로 변경한다면,
- 삼항 연산자가 필요 없어집니다. 단순 boolean값을 반환해도 됩니다
- boolean을 사용하면 타입 체크도 간편합니다
- `checkIsOnOff` 라는 함수명도 `isOn`, `isVisible`, `isActivated`와 같이 더 단순한 형태로 바꿀 수 있습니다

```suggestion
  const isVisible = () =>
    inputRef.current && inputRef.current.value.length && status !== 'default'
```

Reply:
확실히 더욱 깔끔하네요!! 적용해서 커밋 올리겠습니다!